source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -87,8 +140,28 @@ bool Config::isPassThroughTlvTypeNeeded(uint8_t tlv_type) const {
 
 size_t Config::numberOfNeededTlvTypes() const { return tlv_types_.size(); }
 
-bool Config::allowRequestsWithoutProxyProtocol() const {
-  return allow_requests_without_proxy_protocol_;
+bool Config::isVersionAllowed(ProxyProtocolVersion version) const {
+  switch (version) {
+  case ProxyProtocolVersion::NotFound:
+    return allow_requests_without_proxy_protocol_;
+  case ProxyProtocolVersion::V1:
+    return allow_v1_;
+  case ProxyProtocolVersion::V2:
+    return allow_v2_;
+  }
+  return false; // Should never reach here, but needed for windows compiler warning.
```

Comment:
we can use PANIC("should not reach here")

Reply:
Sure, found `PANIC_DUE_TO_CORRUPT_ENUM`, maybe that is a better fit :)